alcs-frontend/src/app/features/board/board.component.spec.ts

```diff
@@ -25,4 +160,228 @@ describe('BoardComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should set the title when provided a board code', async () => {
+    boardEmitter.next([
+      {
+        code: 'boardCode',
+        title: 'boardTitle',
+        isFavourite: false,
+        decisionMaker: '',
+        statuses: [],
+      },
+    ]);
+
+    await fixture.whenStable();
+
+    expect(titleService.setTitle).toHaveBeenCalledTimes(1);
+    expect(titleService.setTitle).toHaveBeenCalledWith('ALCS | boardTitle Board');
+    expect(component.currentBoardCode).toEqual('boardCode');
+  });
+
+  it('should enable covenants when not the vetting board', () => {
+    boardEmitter.next([
+      {
+        code: 'boardCode',
+        title: 'boardTitle',
+        isFavourite: false,
+        decisionMaker: '',
+        statuses: [],
+      },
+    ]);
+
+    expect(component.boardHasCovenant).toBeTruthy();
+  });
+
+  it('should map an application into a card', async () => {
+    boardService.fetchCards.mockResolvedValue({
+      applications: [mockApplication],
+      covenants: [],
+      modifications: [],
+      planningReviews: [],
+      reconsiderations: [],
+    });
+
+    boardEmitter.next([
+      {
+        code: 'boardCode',
+        title: 'boardTitle',
+        isFavourite: false,
+        decisionMaker: '',
+        statuses: [],
+      },
+    ]);
+
+    fixture.detectChanges();
+    await sleep(1);
+
+    expect(component.cards.length).toEqual(1);
+    expect(component.cards[0].title).toEqual('1 (applicant)');
+  });
+
+  it('should map a reconsideration into a card', async () => {
+    boardService.fetchCards.mockResolvedValue({
+      applications: [],
+      covenants: [],
+      modifications: [],
+      planningReviews: [],
+      reconsiderations: [mockRecon],
+    });
+
+    boardEmitter.next([
+      {
+        code: 'boardCode',
+        title: 'boardTitle',
+        isFavourite: false,
+        decisionMaker: '',
+        statuses: [],
+      },
+    ]);
+
+    fixture.detectChanges();
```

Comment:
I think it is worth wrapping this into a helper function with some description regarding sleep. This way once there will be nicer solution we will be able to apply it there

Reply:
Removed fixture.detectChanges as its not needed